React/Views/SafeAreaView/RCTSafeAreaShadowView.m

```diff
@@ -21,12 +22,23 @@ - (void)setLocalData:(RCTSafeAreaViewLocalData *)localData
 
   UIEdgeInsets insets = localData.insets;
 
-  super.paddingLeft = (YGValue){insets.left, YGUnitPoint};
-  super.paddingRight = (YGValue){insets.right, YGUnitPoint};
   super.paddingTop = (YGValue){insets.top, YGUnitPoint};
   super.paddingBottom = (YGValue){insets.bottom, YGUnitPoint};
 
-  [self didSetProps:@[@"paddingLeft", @"paddingRight", @"paddingTop", @"paddingBottom"]];
+  if ([[RCTI18nUtil sharedInstance] doLeftAndRightSwapInRTL]) {
+    if ([[RCTI18nUtil sharedInstance] isRTL]) {
+      super.paddingStart = (YGValue){insets.right, YGUnitPoint};
+      super.paddingEnd = (YGValue){insets.left, YGUnitPoint};
+    } else {
+      super.paddingStart = (YGValue){insets.left, YGUnitPoint};
+      super.paddingEnd = (YGValue){insets.right, YGUnitPoint};
+    }
+    [self didSetProps:@[@"paddingStart", @"paddingEnd", @"paddingTop", @"paddingBottom"]];
+  } else {
+    super.paddingLeft = (YGValue){insets.left, YGUnitPoint};
```

Comment:
paddingLeft and paddingRight always work regardless of `doLeftAndRightSwapInRTL`, so why not always use that?

Reply:
I might be misunderstanding your point here, but I don't think `paddingLeft` and `paddingRight` will always work as we expect.

In the case where `doLeftAndRightSwapInRTL` == true, `RCTProcessMetaPropsPadding` will not set `paddingLeft` and `paddingRight` on the yoga node. Instead it will try to set `paddingStart` and `paddingEnd`, and fallback to `paddingLeft` and `paddingRight` if `paddingStart` and `paddingEnd` are not specified:

```
    YGValue start = metaProps[META_PROP_START].unit == YGUnitUndefined ? metaProps[META_PROP_LEFT] : metaProps[META_PROP_START];
    YGValue end = metaProps[META_PROP_END].unit == YGUnitUndefined ? metaProps[META_PROP_RIGHT] : metaProps[META_PROP_END];
    RCT_SET_YGVALUE(start, YGNodeStyleSetPadding, node, YGEdgeStart);
    RCT_SET_YGVALUE(end, YGNodeStyleSetPadding, node, YGEdgeEnd);
```

This behavior is fine in LTR, but not in RTL (in RTL, `paddingLeft` should be used for `paddingEnd`, `paddingRight` should be used for `paddingStart`).

I totally might be missing something though, so definitely sanity check my logic 😄

Reply:
In other words, yes `paddingLeft` and `paddingRight` would work if `RCTProcessMetaPropsPadding` didn't flip them in RTL. It may be that the right fix is to change `RCTProcessMetaPropsPadding` or not use it in this scenario, but that was a less targeted fix so I opted not to take that path. But your comment about determining isRTL value suggests that such a fix would be a more robust option

Reply:
@cpojer